scripts/mbedtls_dev/psa_information.py

```diff
@@ -0,0 +1,162 @@
+"""Collect information about PSA cryptographic mechanisms.
```

Comment:
The commit "Move PSA information and dependency automation into their own module" should be doing the following:

* Create `psa_information.py` and write some stuff at the top (docstring, copyright notice, imports).
* Move blocks of code from `generate_psa_tests.py` to `psa_information.py`.
* In `generate_psa_tests.py`, add `psa_information.` when referencing functions that have moved to the new module.
* (not in 2.28) update dependencies in makefiles.

b4b5402bfcd4d78c41f98020316295f832add0aa is not doing this correctly: it's adding code from `development`. This commit needs to be reconstructed rather than cherry-picked.

This has consequences, some benign, some not. One bad consequence is that `PSA_WANT_KEY_TYPE_xxx_KEY_PAIR` dependencies have moved to their 3.5 names, which is wrong because those names don't exist in 2.28. This causes some test cases to be never executed.

Reply:
I think I now managed to do the backport properly, sorry for the confusion